### ISSUE TITLE
has no method 'en' Error when trying to translate

### ISSUE THREAD

**Issue by dstollie** [TAPevents/tap-i18n#56]
Hello,

After hours of trying and having like 3 hours sleep I decided to hand my problem over to you guys:

Im getting this error:
```
W20150330-04:32:32.173(2)? (STDERR) TypeError: Object #<Object> has no method 'en'
```
When I want to invoke the TAPi18n.__ method from an smart package

This is how my package.js looks like:

```
Package.onUse(function(api) {
	api.use(['tap:i18n', 'vazco:universe-html-purifier', 'wtfzn:dompurify']);
	api.use(["session"], "client");

	api.addFiles(['package-tap.i18n']);

	api.addFiles([
		'i18n/nl.i18n.json',
		'i18n/en.i18n.json',
		'i18n.js',
		'myapp-common.js'
	]);

	api.export([
		'purify',
		'trans',
		'validEmail'
	]);

});
```
And this is the trans method which uses the __ method
```
trans = function (str, options) {
	if (Meteor.isServer) {
		return TAPi18n.__(str, options, 'nl');
	} else {
		return TAPi18n.__(str, options);
	}
};
```
Hope you guys can help me. I really need my sleep right now ;p

----------------

It looks like the problems is at the server side, client side doesnt give any errors

**Comment by theosp** [comment]
Please replicate the issue and publish it in a github repository.

With that I'll be able to help you much easily.

Thanks,
-Daniel

**Comment by dstollie** [comment]
Hey Daniel,

It's just whenever i use TAPi18n.__('string');

Im getting a:

```
W20150330-08:31:43.888(0)? (STDERR)     at app/server/app.js:5:3                                                                                             
W20150330-08:31:43.890(0)? (STDERR)     at /app/.meteor/local/build/programs/server/boot.js:175:10                                                           
W20150330-08:31:43.891(0)? (STDERR)     at Array.forEach (native)                                                                                            
W20150330-08:31:43.893(0)? (STDERR)     at Function._.each._.forEach (/home/meteor/.meteor/packages/meteor-tool/.1.0.38.ieqxkv++os.linux.x86_64+web.browser+w
eb.cordova/meteor-tool-os.linux.x86_64/dev_bundle/server-lib/node_modules/underscore/underscore.js:79:11)                                                    
W20150330-08:31:43.894(0)? (STDERR)     at /app/.meteor/local/build/programs/server/boot.js:86:5  
```

**Comment by theosp** [comment]
Again,

I want to help you, but I can't replicate this bug, and no one reported it before (last release was more than a month ago).

Replicate it in a project so I can see what's wrong.

-Daniel

**Comment by dstollie** [comment]
Here is the git repo url: https://bitbucket.org/talkkypros/going-talkky.git
The problem is at develop branch and the meteor project is at the "app" folder.

Thanks alot of your help. I really appreciate it!

**Comment by theosp** [comment]
tap:i18n doesn't support at the moment server side translations before startup.

So:
1. See if you can wrap your call with Meteor.startup().
2. Wait for next version that is going to fix this issue (and in the meanwhile help with testing as I keep progressing towards it ;) ).

Thanks,
-Daniel

**Comment by Captain-Banana** [comment]
Note that for pre Meteor.startup() translations to work... **your translation.json files must load before the file where you do a translation!** 

This caught me out. Here's a section from the Meteor docs...

There are several load ordering rules. They are applied sequentially to all applicable files in the application, in the priority given below:

1. HTML template files are always loaded before everything else
2. Files beginning with main. are loaded last
3. Files inside any lib/ directory are loaded next
4. Files with deeper paths are loaded next
5. Files are then loaded in alphabetical order of the entire path


**Load Order Example:**

nav.html
main.html
client/lib/methods.js
client/lib/styles.js
lib/feature/styles.js
lib/collections.js
client/feature-y.js
feature-x.js
client/main.js

**Comment by vipinlahoti** [comment]
Its an old thread but im facing the same issue,
any suggestions here? im on tap:i18n@1.8.2


i18n = function (str, options) {
  if (Meteor.isServer) {
    TAPi18n.__(str, options, 'en');
  } else {
    TAPi18n.__(str, options);
  }
};

**Comment by roelvan** [comment]
@vipinlahoti It means your i18n.json files are being loaded after you call TAPi18n, so no language is set then. Just make sure your json files are being loaded before that. (I had to move mine into the '/lib' folder at the root of my project)

**Comment by jirikrepl** [comment]
I had a similar issue when running server tests.
As it was said before the error probably comes from that the fact JSON files are not loaded by `meteor test` and TAPi18n will crash.

I looked in the TAPi18n source code and mocked necessary functions, so `meteor test` would not fail.

I created this package which is `testOnly`. The code of the package is evaluated before any test files and TAPi18n is mocked.
It works in my case (maybe you will need to tweak it a bit)

https://github.com/jirikrepl/tapi18n-mock

actual two line mock code:D
```
TAPi18n.server_translators.en = (() => {});
TAPi18n.__ = ((x) => x);
```
